src/JJMasterData.Commons/Language/Translate.cs

```diff
@@ -1,18 +1,22 @@
-﻿using System.Collections.Generic;
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Globalization;
 using JJMasterData.Commons.DI;
 
 namespace JJMasterData.Commons.Language;
 
 /// <summary>
-/// Class to i18n 
+/// Translates a key into a localized value.
 /// </summary>
 public static class Translate
 {
-    private static Dictionary<string, Dictionary<string, string>> _resourcesDictionary;
-
-    private static ITranslator Translator => JJService.Translator;
-
+    private static ConcurrentDictionary<string, IDictionary<string, string>> _resourcesDictionary;
```

Comment:
This solves the ugly multi-thread lock problem. (This is what Microsoft uses in IStringLocalizer)